### ISSUE TITLE
resnet 152

### ISSUE THREAD

**Issue by mli** [tornadomeet/ResNet#1]
i'm running the 152 layer model using the default hyper-parameters, will update the results on this issue

current progress

```
INFO:root:Epoch[10] Batch [4850]        Speed: 298.15 samples/sec       Train-accuracy=0.447188
INFO:root:Epoch[10] Batch [4850]        Speed: 298.15 samples/sec       Train-top_k_accuracy_5=0.697656
INFO:root:Epoch[10] Batch [4900]        Speed: 298.05 samples/sec       Train-accuracy=0.445703
INFO:root:Epoch[10] Batch [4900]        Speed: 298.05 samples/sec       Train-top_k_accuracy_5=0.699609
INFO:root:Epoch[10] Batch [4950]        Speed: 297.75 samples/sec       Train-accuracy=0.444453
INFO:root:Epoch[10] Batch [4950]        Speed: 297.75 samples/sec       Train-top_k_accuracy_5=0.695312
INFO:root:Epoch[10] Batch [5000]        Speed: 298.33 samples/sec       Train-accuracy=0.443906
INFO:root:Epoch[10] Batch [5000]        Speed: 298.33 samples/sec       Train-top_k_accuracy_5=0.690469
INFO:root:Epoch[10] Resetting Data Iterator
INFO:root:Epoch[10] Time cost=4299.024
INFO:root:Saved checkpoint to "model/resnet-imagenet-152-0011.params"
INFO:root:Epoch[10] Validation-accuracy=0.507553
INFO:root:Epoch[10] Validation-top_k_accuracy_5=0.765446
INFO:root:Epoch[11] Batch [50]  Speed: 306.66 samples/sec       Train-accuracy=0.444453
INFO:root:Epoch[11] Batch [50]  Speed: 306.66 samples/sec       Train-top_k_accuracy_5=0.694766
INFO:root:Epoch[11] Batch [100] Speed: 298.39 samples/sec       Train-accuracy=0.444609
INFO:root:Epoch[11] Batch [100] Speed: 298.39 samples/sec       Train-top_k_accuracy_5=0.694141
INFO:root:Epoch[11] Batch [150] Speed: 298.08 samples/sec       Train-accuracy=0.439922
INFO:root:Epoch[11] Batch [150] Speed: 298.08 samples/sec       Train-top_k_accuracy_5=0.691719
```

**Comment by tornadomeet** [comment]
great!!
another, i'll verify  RandomSkipResizeIter (https://github.com/dmlc/mxnet/issues/2968 )in cifar10 and resnet-50(from epoch 90), i think it should leads to a better result.

**Comment by mli** [comment]
![figure_1](https://cloud.githubusercontent.com/assets/421857/17953201/4469181c-6a26-11e6-8115-fc22b4cc4827.png)

a large batch size seems help a lot at least for the beginning, here 101 uses 512 while 152 uses 256

**Comment by tornadomeet** [comment]
resnet use many bn layers(especially one bn on the input data for mean-val normalization), and large batch size maybe useful for bn.
looks like resnet-101 is much better than 152 at beginning.

**Comment by tornadomeet** [comment]
the currently resnet-50.log is not strict by `mx.lr_scheduler.MultiFactorScheduler(step=[30*epoch_size, 60*epoch_size, 90*epoch_size], factor=0.1)`, and is `mx.lr_scheduler.MultiFactorScheduler(step=[30*epoch_size, 60*epoch_size, 80*epoch_size], factor=0.1)`  , and at 83 epoch i cancel the scale/ration/hsv augment.  
now i'm retraining according to README.md from epoch 80, and i see it leads better result, when it finished this weekend, i'll update the resnet-50 model and log.

**Comment by tornadomeet** [comment]
@mli   https://github.com/tornadomeet/ResNet/issues/3 this maybe useful~

**Comment by mli** [comment]
![figure_1](https://cloud.githubusercontent.com/assets/421857/17979858/95d3c716-6ab1-11e6-8a65-a253126e9e5d.png)

decreasing lr at epoch 30 is too early. i think we may choose epoch 50 or 60.

**Comment by tornadomeet** [comment]
agree.  deeper network should decrease lr slower, i'll training resnet-18/resnet-34 and choosing epoch 50 or 60.

**Comment by mli** [comment]
![figure_1](https://cloud.githubusercontent.com/assets/421857/18014693/ef6ee162-6b78-11e6-9718-fd68a59cc19c.png)

**Comment by mli** [comment]
i tried "cancel the scale/color/ratio augmentation" at epoch 120, it leads a big decrease of both training and test accuracy.

**Comment by tornadomeet** [comment]
@mli  so what's the acc now?(in resnet-50, when cancel those augmentation, it will improve absolute 1.5%~2.5% of test acc, and > 10% of train acc eventually)

**Comment by tornadomeet** [comment]
does it decrease acc or  increase acc?

**Comment by mli** [comment]
```
INFO:root:Epoch[100] Batch [4900]       Speed: 296.42 samples/sec       Train-accuracy=0.781172
INFO:root:Epoch[100] Batch [4900]       Speed: 296.42 samples/sec       Train-top_k_accuracy_5=0.922891
INFO:root:Epoch[100] Batch [4950]       Speed: 295.91 samples/sec       Train-accuracy=0.776641
INFO:root:Epoch[100] Batch [4950]       Speed: 295.91 samples/sec       Train-top_k_accuracy_5=0.925234
INFO:root:Epoch[100] Batch [5000]       Speed: 295.86 samples/sec       Train-accuracy=0.773828
INFO:root:Epoch[100] Batch [5000]       Speed: 295.86 samples/sec       Train-top_k_accuracy_5=0.920859
INFO:root:Epoch[100] Validation-accuracy=0.767020
INFO:root:Epoch[100] Validation-top_k_accuracy_5=0.929887
```

**Comment by mli** [comment]
canceled 6 augs:

```
INFO:root:Epoch[101] Batch [4950]       Speed: 296.68 samples/sec       Train-accuracy=0.668672
INFO:root:Epoch[101] Batch [4950]       Speed: 296.68 samples/sec       Train-top_k_accuracy_5=0.850156
INFO:root:Epoch[101] Batch [5000]       Speed: 297.75 samples/sec       Train-accuracy=0.661641
INFO:root:Epoch[101] Batch [5000]       Speed: 297.75 samples/sec       Train-top_k_accuracy_5=0.847578
INFO:root:Epoch[101] Resetting Data Iterator
INFO:root:Epoch[101] Time cost=4340.443
INFO:root:Saved checkpoint to "model/resnet-imagenet-152-0102.params"
INFO:root:Epoch[101] Validation-accuracy=0.758889
INFO:root:Epoch[101] Validation-top_k_accuracy_5=0.924705
```

**Comment by tornadomeet** [comment]
it is strange.
- does  `lr` set  correctly ? when cancel those augs, the lr should be the same as before canceling, just like 0.0001 if lr drops 3 times or 0.001 if drops 2 times. 
- when cancel those augs, it should use train_256_q90.rec.

another, i fixed a bug when using `retrain` https://github.com/tornadomeet/ResNet/commit/bcd6a09dafe9d169e9e45e9a762294f45598a37e, maybe you can check that. it will influence the step of lr_scheduler.

**Comment by mli** [comment]
ah, i forget to switch to the 256 rec

**Comment by mli** [comment]
solved by #9

**Comment by liuzhuang13** [comment]
@mli How's the effect of batch size for the final val error? Is 128 better or 256 better? Thanks!

**Comment by mli** [comment]
i guess so but didn't test. also recommend a small batch size if it doesn't
slow down too much

**Comment by liuzhuang13** [comment]
So is 128 better or 256 better? I don't know what you mean by "I guess so"...sorry
From your first figure in this issue it seems a larger batch size is bettter

**Comment by tornadomeet** [comment]
@liuzhuang13 we can get the answer of `whether 128 better or 256 better?` only by experiment, the model trained is batch -size=256, you can tried it using batch-size=128.

**Comment by mli** [comment]
i guess 128 gives better accuracy, because smaller batch size often
converges faster. but as wei said, you need to do the experiment to find
the exact answer.

**Comment by liuzhuang13** [comment]
@tornadomeet @mli  thank you! but unfortunately I don't have that many GPUs to try out. I saw there were both 512 and 256 in the first figure, and 512 seems to be better so I'm curious. Thanks

**Comment by tornadomeet** [comment]
@liuzhuang13 our experiments shows that large batch-size only converge faster at the beginning, but at the end of training, it will converge slower than smaller batch-size, then leads a worse result at the end.

**Comment by liuzhuang13** [comment]
@tornadomeet  Well that's in accordance with my observation on CIFAR dataset, thanks!

**Comment by austingg** [comment]
@tornadomeet  @liuzhuang13  how about using larger batchsize at the begining and  smaller one at the end of the training, just like canceling strong data augmentation at the late epochs?

**Comment by tornadomeet** [comment]
just opposite, need small size at beginning, and large batch-size at the end.
